### PR TITLE
Added faceted/multifaceted filters

### PR DESCRIPTION
**Pull request by 007noon700**

### CODE REVIEW

File: app/Http/Controllers/SearchController.php [hunk 1]
```diff
@@ -43,14 +44,31 @@ public static function apply(Request $filters)
         }
         // Search for a song based on its year.
         if ($filters->has('year')) {
+
             if($filters->input('year')!=null){
-            $song->where('year', $filters->input('year'));
-            $filter['year'] = $filters->input('year');
+                $i = 0;
+                foreach($filters->input('year') as $year) {
```
Comment (james-daily) [inline app/Http/Controllers/SearchController.php:50]: Instead of a loop this can be simplified to `$song->whereIn('year', $filters->input('year'))`

Comment (james-daily) [reply app/Http/Controllers/SearchController.php:50]: Also I can't remember, but if the user only selects a single year do you still get an array?

Also also: server-side validation!

Comment (007noon700) [reply app/Http/Controllers/SearchController.php:50]: Yes, it does appear to still be an array

File: app/Http/Requests/SearchSong.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'year.*' => 'sometimes|nullable|integer|between:1500,3000',
+            'artist' => 'sometimes|nullable|string',
```
Comment (james-daily) [inline app/Http/Requests/SearchSong.php:28]: You could put maximum lengths on these strings, but this is fine.

**Pull request merged**